### PR TITLE
Release 2.19.0

### LINKED ISSUE #3608: `DesktopFileManager::resolve_app_id` should  not return an app id with a `.desktop` file suffix
## Problem
The `app_id` that is returned by `ForeignTopLevelManagerV1` typically ends with a `.desktop` suffix as it currently stands. We did this originally because `g_app_info_get_id` typically returns the app id this way, although it doesn't have to. In cases where the app id is not found with `g_app_info_get_id`, we were appending `.desktop`. We shouldn't do this, as this isn't  necessarily correct.

### PR DESCRIPTION
**Pull request by AlanGriffiths**
# rc1

  * New upstream release 2.19.0

    - ABI summary:
      . miral ABI unchanged at 7
      . mircommon ABI unchanged at 10
      . mircore ABI unchanged at 2
      . miroil ABI unchanged at 5
      . mirplatform ABI bumped to 30
      . mirserver ABI bumped to 61
      . mirwayland ABI unchanged at 5
      . mirplatformgraphics ABI unchanged at 22
      . mirinputplatform ABI unchanged at 9

    - Enhancements:
      . [x11-kiosk] defer fullscreening of applications until they are placed once (#3670)
      . [x11-kiosk] Ensure windows are placed and repainted on X11 "CONFIGURE" (#3619)
      . [x11-kiosk] change enable-x11 default to true
      . Initial atomic-kms platform
      . [Wayland] Add support for xdg_activation_v1 (#3639)
      . Do not default window size (especially to weird values) (#3623)
      . DesktopFileManager::resolve_app_id no longer returns an app id with a .desktop file suffix (Fixes #3608)
      . [Configuration] Split options into global and per-module configuration (#3590)

    - Bugs fixed:
      . Ensure we always send an initial output enter (#3680)
      . Fix stuck-frame-after-mode-switch bug (#3673)
      . Fix rendering of resized XWayland applications with client side decorations (#3587)
      . Aspect ratios shouldn't contain zero (Fixes: #3663)
      . Use PkgConfig to find development headers (#3661)
      . Fix the inotify event handling in ReloadingYamlFileDisplayConfig::auto_reload() (#3636)
      . The `miral::ConfigFile` "Watcher" can be destroyed before the main loop (Fixes: #3612)
      . [xwayland] Don't allow clients to place X11 windows (#3622)
      . Surfaces track scale changes on outputs they appear on. (Fixes: #3552)
      . New attached windows need to be placed (#3676)

----
[Test Plan](https://canonical-mir.readthedocs-hosted.com/latest/how-to/how-to-test-mir-for-a-release/)

### Platforms
|| 24.04 | 24.10 |
|-|-|-|
| gbm-kms |@AlanGriffiths|@AlanGriffiths|
| atomic-kms |@AlanGriffiths|@AlanGriffiths|
| eglstream-kms |@tarek-y-ismail||
| eglstream-kms + gbm-kms hybrid |@tarek-y-ismail||
| x11 |@AlanGriffiths|@AlanGriffiths|
| wayland |@AlanGriffiths|@AlanGriffiths|
| virtual |@AlanGriffiths|@AlanGriffiths|

### Console Providers
|| 24.04 | 24.10 |
|-|-|-|
| vt |@AlanGriffiths|@AlanGriffiths|
| logind |@AlanGriffiths|@AlanGriffiths|
| minimal |@AlanGriffiths||

### Window Manager Examples
|| 24.04 | 24.10 |
|-|-|-|
| --window-manager=floating |@AlanGriffiths|@AlanGriffiths|
| --window-manager=tiling |@AlanGriffiths||
| -kiosk |@AlanGriffiths||

### CODE REVIEW

**Comment by tarek-y-ismail** [comment]
gbm-kms:
- I tested login providers with gbm-kms out of curiosity. Executing the command specified in the logind section just starts `miral-app` without asking me to login. Is this correct?
- When testing the tiling window manager. After opening a terminal with CTRL+ALT+T, trying to open another one crashes the compositor. However, opening the second one through the first works fine?
 
 eglstream-kms + gbm-kms:
 - #3672 and #3598 still occur under the same circumstances
 - `DRI_PRIME` doesn't actually change which GPU is used for rendering.

**Comment by AlanGriffiths** [comment]
> * I tested login providers with gbm-kms out of curiosity. Executing the command specified in the logind section just starts `miral-app` without asking me to login. Is this correct?

Yes.

> * When testing the tiling window manager. After opening a terminal with CTRL+ALT+T, trying to open another one crashes the compositor. However, opening the second one through the first works fine?

I don't see this. Which terminal are you using?

**Comment by AlanGriffiths** [comment]
> > I don't see this. Which terminal are you using?
> 
> This happens with gnome-terminal, kgx, and xterm. But qterminal, weston-terminal don't crash the compositor.

Just trying to figure out what we're doing differently.

I've used `miral-app --window-manager` tiling on `mir:x11` (hosted by Miriway/beta) and `mir:gbm-kms` (on intel+evdi) on 24.04 and 24.10, I don't see this problem with gnome-terminal, kgx or qterminal.

**Comment by tarek-y-ismail** [comment]
For me, I'm running miral-app through a VT with eglstream-kmsa and gbm-kms. Smells of more eglstream-kms trouble?

**Comment by AlanGriffiths** [comment]
> Smells of more eglstream-kms trouble?

Maybe even the same trouble?

**Comment by AlanGriffiths** [comment]
> I'm running miral-app through a VT with eglstream-kms and gbm-kms

Can you test with ppa:mir-team/release and confirm whether this is new or pre-existing?

**Comment by AlanGriffiths** [comment]
> 2.18.3, same thing. 

Not a blocker then.

> I took a look at the logs, and maybe I'm missing something while booting miral-app?

Some of the messages are from gnome terminal, and not relevant to the server crash. The relevant ones are:

> ```
> Mir fatal error: Accessing value of unset optional
> !!! Fatal signal received. Attempting cleanup, but deadlock may occur
> Mir fatal error: Unsupported attempt to continue after a fatal signal: SIGABRT
> ```

Which indicates we have logic wrong somewhere which leads to an abuse of `mir::optional_value`. That should be easy to track down: build locally and reproduce. That should get a usable core. If not a breakpoint on `fatal_error_abort()` will do.

**Comment by AlanGriffiths** [comment]
Huh?! Now I can reproduce this crash (only on 2.19 though). Investigating...

**Comment by AlanGriffiths** [comment]
> Huh?! Now I can reproduce this crash (only on 2.19 though). Investigating...

This is new with 2.19 (since #3623). It's here:

```c++
void TilingWindowManagerPolicy::clip_to_tile(miral::WindowSpecification& parameters, Rectangle const& tile)
{
    auto const displacement = parameters.top_left().value() - tile.top_left;

    auto width = std::min(tile.size.width.as_int()-displacement.dx.as_int(), parameters.size().value().width.as_int());
    auto height = std::min(tile.size.height.as_int()-displacement.dy.as_int(), parameters.size().value().height.as_int());

    parameters.size() = Size{width, height};
}
```

**Comment by AlanGriffiths** [comment]
# rc2

[Test Plan](https://canonical-mir.readthedocs-hosted.com/latest/how-to/how-to-test-mir-for-a-release/)

### Platforms
|| 24.04 | 24.10 |
|-|-|-|
| gbm-kms |||
| atomic-kms |||
| eglstream-kms |@tarek-y-ismail||
| eglstream-kms + gbm-kms hybrid |@tarek-y-ismail||
| x11 |||
| wayland |||
| virtual |||

### Console Providers
|| 24.04 | 24.10 |
|-|-|-|
| vt |||
| logind |||
| minimal |||

### Window Manager Examples
|| 24.04 | 24.10 |
|-|-|-|
| --window-manager=floating |||
| --window-manager=tiling |||
| -kiosk |||

**Comment by Conan-Kudo** [comment]
> mirserver ABI bumped to 61

According to my builds of Mir in Fedora, the mirserver bump happened in 2.18.0, so this is unchanged between 2.18 and 2.19.

File: debian/changelog [hunk 1]
```diff
@@ -1,3 +1,52 @@
+mir (2.19.0) UNRELEASED; urgency=medium
```
Comment (Saviq) [inline debian/changelog:1]: We need a 2.19.1 entry in the changelog…

**Review by Saviq — APPROVED** [review]
Yups. Three times…

**Pull request merged**